### PR TITLE
Sa/adding refresh button

### PR DESCRIPTION
**Pull request by SultanArshad**
Adding pull down refresh (swipe to refresh) on both screens Home & Detail
Screenshot_1658285253

### CODE REVIEW

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/details/DetailViewController.kt [hunk 1]
```diff
@@ -63,6 +65,13 @@ class DetailViewController(bundle: Bundle) : Controller(bundle), KoinComponent,
       router.popCurrentController()
     }
 
+    refreshView.setOnRefreshListener {
+      launch {
+        refreshView.isRefreshing = false
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/details/DetailViewController.kt:70]: shouldn't this be part of state?

**Review by xxfast — CHANGES_REQUESTED** [review]
Do we have any tests for the refresh feature?

File: app-kotlin/build.gradle [hunk 2]
```diff
@@ -54,7 +58,7 @@ dependencies {
 //  Conductor
   implementation 'com.bluelinelabs:conductor:3.1.5'
 
-// Lifecycle-aware Controllers (architecture components):
+//  Lifecycle-aware Controllers (architecture components):
```
Comment (xxfast) [inline app-kotlin/build.gradle:61]: is this intended?

File: app-kotlin/build.gradle [hunk 3]
```diff
@@ -65,10 +69,10 @@ dependencies {
   implementation 'com.squareup.moshi:moshi-kotlin:1.13.0'
 
 //  Kotlin Coroutines
-  implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.3'
-  testImplementation 'org.jetbrains.kotlinx:kotlinx-coroutines-test:1.6.3'
+  implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.4'
+  testImplementation 'org.jetbrains.kotlinx:kotlinx-coroutines-test:1.6.4'
 
-//  Picasso
+//  picasso
```
Comment (xxfast) [inline app-kotlin/build.gradle:75]: is this intended?

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/details/DetailViewController.kt [hunk 4]
```diff
@@ -63,6 +65,13 @@ class DetailViewController(bundle: Bundle) : Controller(bundle), KoinComponent,
...
+        viewModel.getPokemonDetail(pokemon.id)
+      }
+    }
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/details/DetailViewController.kt:73]: this is a presenter responsibility, yeah?

**Review by xxfast — CHANGES_REQUESTED** [review]
Keep this PR limited to adding the refresh functionality. Let's make another PR to refactor the list screen to introduce the new architecture

**Comment by SultanArshad** [comment]
Ok i am looking into this

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/details/DetailViewController.kt (hunk 4 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/screens/details/DetailViewController.kt]: yes i had update the code according to feature you can see this update in UPDATING architecture branch

File: app-kotlin/build.gradle (hunk 3 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:72]: not intended is there any way to avoid such single space change in commit ?

**Comment by SultanArshad** [comment]
Updated code specific to Swipe to refresh  feature

**Pull request closed without merging**